src/model.ts

```diff
@@ -462,3 +463,61 @@ export class SpecQueryModel {
     this._rankingScore[rankingName] = score;
   }
 }
+
+export class SpecQueryModelGroup {
+  private _name: string;
+  private _path: string;
+  private _items: (SpecQueryModel | SpecQueryModelGroup)[];
+  private _groupBy: GroupBy;
+  private _orderGroupBy: string;
+
+  constructor(name: string = '', path: string = '', items: (SpecQueryModel | SpecQueryModelGroup)[] = [],
+              groupBy: GroupBy = undefined, orderGroupBy: string = undefined) {
+    this._name = name;
+    this._path = path;
+    this._items = items;
+    this.groupBy = groupBy;
+    this._orderGroupBy = orderGroupBy;
+  }
+
+  public getTopSpecModel(g: SpecQueryModelGroup): SpecQueryModel {
+    const topItem = g._items[0];
+    if (topItem instanceof SpecQueryModelGroup) {
+      return this.getTopSpecModel(topItem);
+    } else {
+      return topItem;
+    }
+  }
+
+  public get name() {
+    return this._name;
+  }
+
+  public set name(name: string) {
```

Comment:
Do we need all of these setters?  Is there way to refactor code such that a lot of these things get defined when the constructor is called? 

Generally, I feel like if we have both getter and setters for everything, why don't we make all of them just a public members?